pkg/kine/sqllog/sqllog.go

```diff
@@ -71,6 +72,7 @@ type SQLLog struct {
 	d           Dialect
 	broadcaster broadcaster.Broadcaster[[]*server.Event]
 	notify      chan int64
+	currentRev  int64
```

Comment:
This seems like a mistake. Caching something we do not have control over will result in weird errors along the way. 

For example, imagine that we reject a watch request asking for version `x`. Now what happens if `x` was compacted, but the polling interval didn't expire yet? We will either return a stale progress report or return an error with a version which sound valid but it isn't.

It also sounds like premature optimization: do we have data hinting that fetching the "real" revision would be expensive?

Reply:
Our friends at kine [argued that](https://github.com/k3s-io/kine/pull/268):
```
Optimize CurrentRevision checks by sharing current revision marker from poll loop
Progress notifications query the current revision frequently; this can be optimized by exposing the poll loop's current revision once the loop has been started.
```
Our ProgressNotify interval is 5 seconds and there is no hard limit on the amount of watchers you can have.
If you think about deployments with 150,000 pods (number comes from [here](https://kubernetes.io/docs/setup/best-practices/cluster-large/)), would it still be cheap to get this read?

Reply:
In MM agreed on removing caching.